common/src/com/elikill58/negativity/common/protocols/AimBot.java

```diff
@@ -46,6 +46,69 @@ public AimBot() {
 		super(CheatKeys.AIM_BOT, CheatCategory.COMBAT, Materials.TNT, CheatDescription.VERIF);
 	}
 
+	// this check seems stupid, although it's a must-have
+	// every decent anti cheat needs this as it fixes most old scaffolds/killauras
+	@Check(name = "invalid", conditions = CheckConditions.SURVIVAL, description = "Checks for an invalid rotation", ignoreCancel = true)
+	public void invalidP(PacketReceiveEvent e, NegativityPlayer np) {
+		if (!e.hasPlayer())
+			return;
+		PacketType type = e.getPacket().getPacketType();
+		if (type.isFlyingPacket()) {
+			NPacketPlayInFlying flying = (NPacketPlayInFlying) e.getPacket().getPacket();
+			if (!flying.hasLook)
+				return;
+			// easiest aim check ever?
+			// and yes, it seems stupid, but many clients do not follow this
+			// especially old scaffolds will fail this check
+			final double maxP = np.isOnLadders ? 91.11d : 90d;
+			final double absolutePitch = Math.abs(flying.pitch);
+			if (absolutePitch > maxP) {
+				Negativity.alertMod(ReportType.WARNING, np.getPlayer(), this, 100, "invalid", "pitch: "
+						+ String.format("%.2f", absolutePitch) + "/" + maxP);
+			}
+		}
+	}
+
+	// many killauras use a constant pitch in order to bypass the GDC check
+	// this check will fight against that and fail these killauras
+	@Check(name = "ratio", conditions = CheckConditions.SURVIVAL, description = "Checks for invalid rotation ratios", ignoreCancel = true)
+	public void ratio(PacketReceiveEvent e, NegativityPlayer np) {
+		if (!e.hasPlayer())
+			return;
+		PacketType type = e.getPacket().getPacketType();
+		if (type.isFlyingPacket()) {
+			NPacketPlayInFlying flying = (NPacketPlayInFlying) e.getPacket().getPacket();
+			if (!flying.hasLook)
+				return;
+			if (!np.isAttacking)
+				return;
+			double deltaYaw = flying.yaw - np.doubles.get(getKey(), "last-yaw-streak", 0.0);
+			double deltaPitch = flying.pitch - np.doubles.get(getKey(), "last-pitch-streak", 0.0);
+			double lastDeltaPitch = np.doubles.get(getKey(), "last-delta-pitch-streak", 0.0);
+			final double difference = Math.abs(deltaPitch - lastDeltaPitch);
+			final double absoluteDeltaYaw = Math.abs(deltaYaw);
+			if (difference < 0.005 && absoluteDeltaYaw > .65) {
+				// increment streak
+				np.ints.set(getKey(), "ratio-streak", np.ints.get(getKey(), "ratio-streak", 0) + 1);
+
+				final int streak = np.ints.get(getKey(), "ratio-streak", 0);
+
+				if (streak > 7) {
+					Negativity.alertMod(ReportType.WARNING, np.getPlayer(), this, 100, "ratio", "absYaw: "
```

Comment:
Same thing for setBack. But, why always use 100% as reliability, and not make quick math to let it be low ? And/or do math for alert amount ?